hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -233,6 +233,10 @@ protected Dialect() {
 		registerHibernateType( Types.CLOB, StandardBasicTypes.CLOB.getName() );
 		registerHibernateType( Types.REAL, StandardBasicTypes.FLOAT.getName() );
 
+		if(supportsPartitionBy()) {
+			registerKeyword( "PARTITION" );
+		}
+
```

Comment:
That was my original approach, but then I came to needing this info as a `DialectCheck`. I can add the keyword registration into every specific `Dialect` implementation, but then I'll have to use  the deprecated `getKeywords()` method because `IdentifierHelper` does not expose the underlying registered keywords.